src/vendorpromo/views.py

```diff
@@ -105,6 +107,158 @@ class AffiliateDeleteView(LoginRequiredMixin, DeleteView):
+    now = timezone.now()
+    promo_offer = Offer()
+    promo_offer.name = promo_campaign.name
+    promo_offer.start_date = now
+    promo_offer.site = promo_campaign.site
+    promo_offer.is_promotional = True
+    promo_offer.save()
+
+    for product in products.all():
+        promo_offer.products.add(product)
+
+    price = Price()
+    price.offer = promo_offer
+    price.cost = cost
+    price.start_date = now
+    price.save()
+
+    return promo_offer
+
+
+def valid_form_save_promotional_campaign(request, form):
+    promo_campaign = form.save(commit=False)
+    site = get_site_from_request(request)
+    promo_campaign.site = site
+    promo_campaign.is_percent_off = form.cleaned_data['is_percent_off']
+    promo_campaign.applys_to = create_promo_offer(promo_campaign, form.cleaned_data['applys_to'], form.cleaned_data['discount_value'])
+    promo_campaign.save()
+
+
+class PromotionalCampaignListView(LoginRequiredMixin, TableFilterMixin, ListView):
+    template_name = "vendorpromo/promotional_campaign_list.html"
+    model = PromotionalCampaign
+    paginate_by = 100
+
+    def search_filter(self, queryset):
+        search_value = self.request.GET.get('search_filter')
+        return queryset.filter(Q(pk__icontains=search_value)
+                               | Q(name__icontains=search_value)
+                               | Q(description__icontains=search_value))
+    
+    def get_paginated_by(self, queryset):
+        if 'paginate_by' in self.request.kwargs:
+            return self.kwargs['paginate_by']
+        return self.paginate_by
+    
+    def get_queryset(self):
+        site = get_site_from_request(self.request)
+        queryset = super().get_queryset().filter(site=site)
+
+        return queryset.order_by('name')
+
+
+class PromotionalCampaignCreateView(LoginRequiredMixin, CreateView):
+    template_name = 'vendorpromo/promotional_campaign_detail.html'
+    model = PromotionalCampaign
+    form_class = PromotionalCampaignForm
+    success_url = reverse_lazy('promotional-campaign-list')
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['site'] = get_site_from_request(self.request)
+        return kwargs
+
+    def form_valid(self, form):
+        valid_form_save_promotional_campaign(self.request, form)
+        return redirect(self.success_url)
+
+
+class PromotionalCampaignUpdateView(LoginRequiredMixin, UpdateView):
+    template_name = 'vendorpromo/promotional_campaign_detail.html'
+    model = PromotionalCampaign
+    form_class = PromotionalCampaignForm
+    slug_field = 'uuid'
+    slug_url_kwarg = 'uuid'
+    success_url = reverse_lazy('promotional-campaign-list')
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['site'] = get_site_from_request(self.request)
+        return kwargs
+
+    def form_valid(self, form):
+        valid_form_save_promotional_campaign(self.request, form)
```

Comment:
this seems like it would create a new offer every time update is called (via create_promo_offer)... should probably write a separate function for updating applys_to, or edit the current one to be more of a get_or_create

Reply:
So glad you caught that. Thanks